### PR TITLE
Moved to Servlet API 6.0

### PR DESCRIPTION
**Pull request by ethauvin**
And also added more Tomcat server tests.

### CODE REVIEW

File: src/main/java/rife/engine/CookieBuilder.java [hunk 1]
```diff
@@ -43,17 +43,6 @@ public Cookie cookie() {
         return cookie_;
     }
 
-    /**
```
Comment (gbevin) [inline src/main/java/rife/engine/CookieBuilder.java:46]: Why did you remove the comment and version methods in the builder?

File: src/test/java/rife/engine/ContextInfo.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Erik C. Thauvin (https://erik.thauvin.net/)
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ */
+package rife.engine;
+
+public class ContextInfo {
```
Comment (gbevin) [inline src/test/java/rife/engine/ContextInfo.java:7]: I don't understand the user of this class, it seems to just store the information from `Context` and that is then used to access it again. Why not simply use the context methods directly?

File: src/test/java/rife/engine/TestEngine.java [hunk 3]
```diff
@@ -542,13 +752,6 @@ void testGenerateEmptyFormPrefix()
         }
     }
 
-    class Routes extends Router {
```
Comment (gbevin) [inline src/test/java/rife/engine/TestEngine.java:545]: This class is located here because it's used by the test method that follows it directly.

File: src/main/java/rife/engine/CookieBuilder.java (hunk 1 above)
Comment (ethauvin) [reply src/main/java/rife/engine/CookieBuilder.java:46]: Deprecated.

Comment (gbevin) [reply src/main/java/rife/engine/CookieBuilder.java:46]: It's still part of the API and since the builder forwards this, I think we should keep it in there

Comment (ethauvin) [reply src/main/java/rife/engine/CookieBuilder.java:46]: IT won't be there in the next API version, so I don't think we should.

File: src/test/java/rife/engine/ContextInfo.java (hunk 2 above)
Comment (ethauvin) [reply src/test/java/rife/engine/ContextInfo.java]: Me neither, maybe I was thinking I was going to reuse it somewhere else, or something.

File: src/test/java/rife/engine/TestEngine.java (hunk 3 above)
Comment (ethauvin) [reply src/test/java/rife/engine/TestEngine.java:545]: Your codestyle didn't think so.

File: src/main/java/rife/engine/CookieBuilder.java (hunk 1 above)
Comment (ethauvin) [reply src/main/java/rife/engine/CookieBuilder.java:46]: To be clear, it's [deprecated and scheduled for removal](https://jakarta.ee/specifications/servlet/6.0/apidocs/jakarta.servlet/jakarta/servlet/http/cookie#setComment(java.lang.String)).

File: src/test/java/rife/engine/ContextInfo.java (hunk 2 above)
Comment (ethauvin) [reply src/test/java/rife/engine/ContextInfo.java:7]: Well, tests failed. I remembered that I did it that way because it is set in a lambda and must be a final object.
